### DISCUSSION TITLE
`toString` and `toString()`

### DISCUSSION THREAD

**Discussion by valencik** [typelevel/case-insensitive#309] [Q&A] [answered]
Hi 👋 

We were using this repo as inspiration for another project and copied the `override val toString` in the following:
```scala
final class CIString private (override val toString: String)
```
from [CIString.scala](https://github.com/typelevel/case-insensitive/blob/d523bdae63daa37ec1bf72cd45864993fa659c9c/core/src/main/scala/org/typelevel/ci/CIString.scala#L55)

The end result is that a `CIString` has both a `toString` val and a `toString()` method.
I admit, I was a tiny bit surprised by this.
I'm curious what the rationale is for having both?

**Comment by armanbilge** [comment, marked as answer]
Hmm, I'm pretty sure those are the same thing? In Scala `val`s are encoded as methods.

```java
/*
 * Decompiled with CFR 0.151.
 */
public final class CIString {
    private final String toString;

    private CIString(String toString) {
        this.toString = toString;
    }

    public String toString() {
        return this.toString;
    }
}
```
